### PR TITLE
Streams: Check constructor and method names

### PR DESCRIPTION
**Pull request by ricea**
Check the .name properties of constructors and methods for the Streams API.

Also add properties tests for ReadableByteStreamController,
ReadableStreamBYOBReader and ReadableStreamBYOBRequest.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17861)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/269548124)

File: streams/readable-byte-streams/properties.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      controller = c;
+    },
+    type: 'bytes'
+  });
+  const proto = Object.getPrototypeOf(controller);
+
+  assert_array_equals(Object.getOwnPropertyNames(proto).sort(), properties);
+
+  for (const m of methods) {
+    const propDesc = Object.getOwnPropertyDescriptor(proto, m);
```
Comment (domenic) [inline streams/readable-byte-streams/properties.js]: Lots of duplication here and elsewhere, but meh...

**Review by domenic — APPROVED** [review]
Setting aside pre-existing duplication issues, LGTM.

File: streams/readable-byte-streams/properties.js (hunk 1 above)
Comment (ricea) [reply streams/readable-byte-streams/properties.js]: I know. I wanted to use the structure I used in [writable-streams/properties.js](https://github.com/w3c/web-platform-tests/blob/master/streams/writable-streams/properties.js) but the need to fetch ReadableStreamBYOBRequest asynchronously meant it wouldn't work directly. So then I just decided to go ahead and land this before the scope creep got any worse.

**Comment by ricea** [comment]
I'm a bit worried that the Travis failures for Safari and Edge might mean the test never completes on those browsers. Does w-p-t have a way of dealing with that if I merge this change?

**Comment by domenic** [comment]
My interpretation was that the WPT infra is just broken and not running tests in any browsers. (Since Chrome is not working either.) Let me ask in #testing IRC to see what's up.

**Comment by domenic** [comment]
gsnedders helpfully said

> Chrome looks like the same old probably-ChromeDriver issue, Safari/Edge look like Sauce being down again

So I think we can merge this.

**Pull request merged**